lib/modules/manager/npm/post-update/index.ts

```diff
@@ -322,6 +322,24 @@ async function resetNpmrcContent(
   }
 }
 
+export function fuzzyMatchAdditionalYarnrcYml<
+  T extends { npmRegistries?: Record<string, unknown> },
+>(additionalYarnRcYml: T, existingYarnrRcYml: T): T {
+  return {
+    ...additionalYarnRcYml,
+    npmRegistries: Object.entries(additionalYarnRcYml.npmRegistries ?? {})
+      .map(([k, v]) => {
+        const key =
+          Object.keys(existingYarnrRcYml.npmRegistries ?? {}).find(
+            // match without trailing slashes
+            (x) => x.replace(/\/$/, '').endsWith(k.replace(/\/$/, '')),
+          ) ?? k;
```

Comment:
```suggestion
    keys.get(k) ?? k;
```


create a map outside the map loop, so we do this once